src/sid.cpp

```diff
@@ -155,6 +156,13 @@ byte Sid::shape(int voice) const { return registers[7 * voice + CONTROL] & 0xf0;
 
 void Sid::set_filter_mode(uint8_t value) {
 	write_mask(registers[FILTER_MODE_VOLUME], HIGHPASS | BANDPASS | LOWPASS, value);
+
```

Comment:
This is not needed. `set_filter_mode` uses `write_mask` which only overwrites the high/band/lowpass bit,s but does not overwrite the volume bits in the same register.

Instead, add a `Sid::set_volume` function which uses `write_mask` (or `bitWrite` if you prefer) to set the volume bits of the same register.

Reply:
got it